rmw_fastrtps_cpp/src/functions.cpp

```diff
@@ -1014,6 +992,12 @@ extern "C"
         subscriberParam.topic.topicKind = NO_KEY;
         subscriberParam.topic.topicDataType = type_name;
         subscriberParam.topic.topicName = topic_name;
+        subscriberParam.historyMemoryPolicy = PREALLOCATED_WITH_REALLOC_MEMORY_MODE;
+
+        Locator_t multicast_locator;
+        multicast_locator.set_IP4_address("239.255.0.1");
+        multicast_locator.port = 7600;
+        subscriberParam.multicastLocatorList.push_back(multicast_locator);
```

Comment:
Restarting my VM addressed the issue... That might be a case we want to handle more gracefully in the future.